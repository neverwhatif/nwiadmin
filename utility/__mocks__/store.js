let jwt = 'jwt';

module.exports = {
    getItem: () => jwt,
    setItem: (key, value) => `${key}:${value}`,
    removeItem: () => null,
    mockUnsetItem: () => {
        jwt = undefined;
    },
    mockResetItem: () => {
        jwt = 'jwt';
    },
};
