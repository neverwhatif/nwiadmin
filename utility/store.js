const tryUseLocal = () => {
    try {
        localStorage.setItem('a', 'b');
    } catch (exception) {
        return false;
    }

    localStorage.removeItem('a');
    return true;
};

const useLocal = tryUseLocal();

const setCookie = (key, value) => {
    if (typeof document === 'undefined') {
        return;
    }
    document.cookie = `${key}=${value}; path=/`;
};

const getCookie = (key) => {
    if (typeof document === 'undefined') {
        return null;
    }

    const nameEQ = `${key}=`;
    const ca = document.cookie.split(';');

    for (let i = 0; i < ca.length; i += 1) {
        let c = ca[i];
        while (c.charAt(0) === ' ') c = c.substring(1, c.length);
        if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
    }

    return null;
};

const removeCookie = (key) => {
    if (typeof document === 'undefined') {
        return;
    }
    document.cookie = `${key}=; expires=Thu, 01 Jan 1970 00:00:01 GMT;`;
};

const getItem = (key) => {
    if (useLocal && window) {
        return window.localStorage.getItem(key);
    }
    return getCookie(key);
};

const getObject = key => JSON.parse(getItem(key));

const setItem = (key, value) => {
    if (useLocal && window) {
        return window.localStorage.setItem(key, value);
    }
    return setCookie(key, value);
};

const setObject = (key, object) => setItem(key, JSON.stringify(object));

const removeItem = (key) => {
    if (useLocal && window) {
        return window.localStorage.removeItem(key);
    }
    return removeCookie(key);
};

export default {
    getItem,
    getObject,
    setItem,
    setObject,
    removeItem,
};
