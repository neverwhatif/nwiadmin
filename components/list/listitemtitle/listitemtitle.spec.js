import React from 'react';
import { shallow } from 'enzyme';

import store from 'nwiadmin/utility/store';
import ListItemTitle from './index';

jest.mock('nwiadmin/components/link', () => 'Link');
jest.mock('nwiadmin/components/reference', () => 'Reference');

describe('ListItemTitle', () => {
    const component = shallow(<ListItemTitle className="lorem" title="Lorem" />);
    const componentLink = shallow(<ListItemTitle className="lorem" title="Lorem" to="ipsum" />);
    const componentRef = shallow(<ListItemTitle className="lorem" title="Lorem" reference="ipsum" />);
    const componentLinkRef = shallow(<ListItemTitle className="lorem" title="Lorem" to="ipsum" reference="ipsum" />);
    const componentPreload = shallow(<ListItemTitle className="lorem" title="Lorem" to="ipsum" shouldInitPreload />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentLink).toMatchSnapshot();
        expect(componentRef).toMatchSnapshot();
        expect(componentLinkRef).toMatchSnapshot();
        expect(componentPreload).toMatchSnapshot();
    });
    it('should render class', () => {
        expect(component.hasClass('lorem')).toBe(true);
    });
    it('should render link', () => {
        expect(component.find('Link').length).toBe(0);
        expect(componentLink.find('Link').length).toBe(1);
        expect(componentLink.find('Link').prop('to')).toBe('ipsum');
    });
    it('should render reference', () => {
        expect(component.find('Reference').length).toBe(0);
        expect(componentRef.find('Reference').length).toBe(1);
        expect(componentRef.find('Reference').prop('title')).toBe('Lorem');
        expect(componentRef.find('Reference').prop('reference')).toBe('ipsum');
        expect(componentRef.find('Reference').prop('isLinked')).toBe(true);
    });
    it('should render link reference', () => {
        expect(componentLinkRef.find('Link').length).toBe(1);
        expect(componentLinkRef.find('Reference').length).toBe(1);
        expect(componentLinkRef.find('Reference').prop('isLinked')).toBe(false);
    });
    it('should preload', () => {
        const spy = jest.spyOn(store, 'setItem');

        componentPreload.find('Link').simulate('click');
        expect(spy).toHaveBeenCalled();
    });
});
