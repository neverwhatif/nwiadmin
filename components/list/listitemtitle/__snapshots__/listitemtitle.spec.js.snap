// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemTitle should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItemTitle
    className="lorem"
    reference={null}
    shouldInitPreload={false}
    title="Lorem"
    to={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "lorem",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "lorem",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ListItemTitle should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItemTitle
    className="lorem"
    reference={null}
    shouldInitPreload={false}
    title="Lorem"
    to="ipsum"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        to="ipsum"
      >
        Lorem
      </Link>,
      "className": "lorem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "onClick": [Function],
        "to": "ipsum",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "Link",
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          to="ipsum"
        >
          Lorem
        </Link>,
        "className": "lorem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "onClick": [Function],
          "to": "ipsum",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "Link",
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ListItemTitle should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItemTitle
    className="lorem"
    reference="ipsum"
    shouldInitPreload={false}
    title="Lorem"
    to={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Reference
        isLinked={true}
        reference="ipsum"
        title="Lorem"
      />,
      "className": "lorem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "isLinked": true,
        "reference": "ipsum",
        "title": "Lorem",
      },
      "ref": null,
      "rendered": null,
      "type": "Reference",
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Reference
          isLinked={true}
          reference="ipsum"
          title="Lorem"
        />,
        "className": "lorem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "isLinked": true,
          "reference": "ipsum",
          "title": "Lorem",
        },
        "ref": null,
        "rendered": null,
        "type": "Reference",
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ListItemTitle should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItemTitle
    className="lorem"
    reference="ipsum"
    shouldInitPreload={false}
    title="Lorem"
    to="ipsum"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        to="ipsum"
      >
        <Reference
          isLinked={false}
          reference="ipsum"
          title="Lorem"
        />
      </Link>,
      "className": "lorem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Reference
          isLinked={false}
          reference="ipsum"
          title="Lorem"
        />,
        "onClick": [Function],
        "to": "ipsum",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "isLinked": false,
          "reference": "ipsum",
          "title": "Lorem",
        },
        "ref": null,
        "rendered": null,
        "type": "Reference",
      },
      "type": "Link",
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          to="ipsum"
        >
          <Reference
            isLinked={false}
            reference="ipsum"
            title="Lorem"
          />
        </Link>,
        "className": "lorem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Reference
            isLinked={false}
            reference="ipsum"
            title="Lorem"
          />,
          "onClick": [Function],
          "to": "ipsum",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "isLinked": false,
            "reference": "ipsum",
            "title": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "Reference",
        },
        "type": "Link",
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ListItemTitle should render 5`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItemTitle
    className="lorem"
    reference={null}
    shouldInitPreload={true}
    title="Lorem"
    to="ipsum"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        to="ipsum"
      >
        Lorem
      </Link>,
      "className": "lorem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "onClick": [Function],
        "to": "ipsum",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "Link",
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          to="ipsum"
        >
          Lorem
        </Link>,
        "className": "lorem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "onClick": [Function],
          "to": "ipsum",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "Link",
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
